ui/src/slicer/SlicerArrows.ts

```
@@ -309,45 +323,194 @@ export default class SlicerArrows {
+          directionVector,
+          ARROW_LENGTH + ARROW_TIP_LENGTH - ARROW_TIP_OFFSET,
+          new Cartesian3(),
+        ),
+        new Cartesian3(),
+      );
+    }, false);
+  }
+
   highlightArrow(position) {
     const pickedObject = this.viewer.scene.pick(position);
-    const isModelPicked =
-      pickedObject && pickedObject.id && pickedObject.id.model;
-    if (
-      isModelPicked &&
-      pickedObject.id.properties &&
-      pickedObject.id.properties.side
-    ) {
-      this.highlightedArrow = pickedObject.id;
-      this.viewer.canvas.style.cursor = 'pointer';
-      // @ts-ignore 2322
-      this.highlightedArrow.model.color = Color.YELLOW;
+    if (pickedObject?.id.cylinder) {
+      const entry = Object.values(this.arrows).find(
+        (entry) =>
+          entry.shaft === pickedObject.id ||
+          entry.topCone === pickedObject.id ||
+          entry.bottomCone === pickedObject.id,
+      );
+      if (entry) {
+        this.highlightedArrow = entry;
+        this.viewer.canvas.style.cursor = 'pointer';
+        // @ts-expect-error 2322
+        this.highlightedArrow.shaft.cylinder.material = Color.YELLOW;
+        // @ts-expect-error 2322
+        this.highlightedArrow.topCone.cylinder.material = Color.YELLOW;
+        // @ts-expect-error 2322
+        this.highlightedArrow.bottomCone.cylinder.material = Color.YELLOW;
+      }
     } else {
       this.unhighlightArrow();
     }
   }
 
   unhighlightArrow() {
-    if (this.highlightedArrow?.model) {
-      // @ts-ignore 2322
-      this.highlightedArrow.model.color = SLICING_GEOMETRY_COLOR;
+    if (this.highlightedArrow) {
+      // @ts-expect-error 2322
+      this.highlightedArrow.shaft.cylinder.material = SLICING_GEOMETRY_COLOR;
+      // @ts-expect-error 2322
+      this.highlightedArrow.topCone.cylinder.material = SLICING_GEOMETRY_COLOR;
+      // @ts-expect-error 2322
+      this.highlightedArrow.bottomCone.cylinder.material =
+        SLICING_GEOMETRY_COLOR;
```

Instead of suppressing the error here, you should probably wrap the values into `new ConstantProperty(<value>)`.